File: Make Human Unity/Assets/MakeHuman/Icons/KsSoft/Editor/Multilingual/tools/protocol/t_Message.py
from CVariable import CReadVariable,CWriteVariable
from Vector import *
from t_MessageArg import	t_MessageArg
#/==========================================================================
#*!
#	@brief	t_Message
#/
class t_Message:
	#/==========================================================================
	#*!
	#	@brief	Member
	#/
	m_mSheetId = 0	#U32
	m_mId = 0	#U32
	m_aArg = []	#t_MessageArg
	m_aString = []	#std::string
	#/==========================================================================
	#*!
	#	@brief	Constructor
	#/
	def __init__(self):
		self.clear()

	#/==========================================================================
	#*!
	#	@brief	Accessor
	#/
	def clear(self):
		self.m_mSheetId = 0
		self.m_mId = 0
		self.m_aArg = []
		self.m_aString = []

	def read(self,cVariable):
		self.m_mSheetId = cVariable.getU32()
		self.m_mId = cVariable.getU32()
		n = cVariable.getU8()
		if (n > 255) :
			cVariable.error('array size error in m_aArg')
			return False
		self.m_aArg = [None]*n
		for i in range(n):
			self.m_aArg[i] = t_MessageArg()
			if (not self.m_aArg[i].read(cVariable)):
				return False
		n = cVariable.getU8()
		if (n > 255) :
			cVariable.error('array size error in m_aString')
			return False
		self.m_aString = [0]*n
		for i in range(n):
			self.m_aString[i] = cVariable.getString(255)
		return True

	def write(self,cVariable):
		cVariable.putU32(self.m_mSheetId)
		cVariable.putU32(self.m_mId)
		if (len(self.m_aArg) > 255) :
			cVariable.error('array size error in m_aArg')
			return False
		n = len(self.m_aArg)
		cVariable.putU8(n)
		for i in range(n):
			if (not self.m_aArg[i].write(cVariable)):
				return False
		if (len(self.m_aString) > 255) :
			cVariable.error('array size error in m_aString')
			return False
		n = len(self.m_aString)
		cVariable.putU8(n)
		for i in range(n):
			cVariable.putString(self.m_aString[i],255)
		return True

File: Assets/MakeHuman/Icons/KsSoft/Plugins/Protocol/t_Message.cs
//==============================================================================================
/*!
	@file  KsSoft Protocol
	
	(counter SJIS string 京)
*/
//==============================================================================================
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

using KS;
//==========================================================================
/*!
	@brief	t_Message
*/
public class t_Message : ISerializable {
	//==========================================================================
	/*!
		@brief	Member
	*/
	public uint	m_mSheetId;
	public uint	m_mId;
	public t_MessageArg[]	m_aArg;
	public string[]	m_aString;
	//==========================================================================
	/*!
		@brief	Constructor
	*/
	public t_Message() {
		clear();
	}
	//==========================================================================
	/*!
		@brief	Accessor
	*/
	public void clear() {
		m_mSheetId = 0;
		m_mId = 0;
		m_aArg = new t_MessageArg[0];
		m_aString = new string[0];
	}
	public bool read(CReadVariable cVariable) {
		try {
			m_mSheetId = cVariable.getU32();
			m_mId = cVariable.getU32();
			{
				int n = (int) cVariable.getU8();
				if (n > 255) {
					cVariable.error("array size error in m_aArg");
					return false;
				}
				m_aArg = new t_MessageArg [n];
				for(int i = 0;i < n;++i) {
					m_aArg[i] = new t_MessageArg();
					if (!m_aArg[i].read(cVariable)) return false;
				}
			}
			{
				int n = (int) cVariable.getU8();
				if (n > 255) {
					cVariable.error("array size error in m_aString");
					return false;
				}
				m_aString = new string [n];
				for(int i = 0;i < n;++i) {
					cVariable.getString(ref m_aString[i],255);
				}
			}
		} catch (System.Exception e) {
			Debug.LogError(e);return false;
		}
		return true;
	}
	public bool write(CWriteVariable cVariable) {
		cVariable.put(m_mSheetId);
		cVariable.put(m_mId);
		{	
				if (m_aArg.Length > 255) {
					cVariable.error("array size error in m_aArg");
					return false;
				}
			int n = m_aArg.Length;
			cVariable.put((byte) n);
			for(int i = 0;i < n;++i) {
				if (!m_aArg[i].write(cVariable)) return false;

			}
		}
		{	
				if (m_aString.Length > 255) {
					cVariable.error("array size error in m_aString");
					return false;
				}
			int n = m_aString.Length;
			cVariable.put((byte) n);
			for(int i = 0;i < n;++i) {
				cVariable.put(ref m_aString[i],255);

			}
		}
		return true;
	}
};

